pipelines/controller/src/main/java/org/openmrs/analytics/PipelineManager.java

```diff
@@ -304,6 +307,85 @@ synchronized void runIncrementalPipeline()
     }
   }
 
+  /**
+   * This method checks upon Controller start checks on Thrift sever to create resource tables if
+   * they don't exist. There is a @PostConstruct method present in this class which is initDwhStatus
+   * and the reason below code has not been added because dataProperties.getThriftserverHiveConfig()
+   * turns out to be null when used by
+   * DatabaseConfiguration.createConfigFromFile(dataProperties.getThriftserverHiveConfig()).
+   */
+  @Override
+  public void onApplicationEvent(ApplicationReadyEvent applicationReadyEvent) {
+    if (!dataProperties.isCreateHiveResourceTables()) {
+      return;
+    }
+
+    DatabaseConfiguration dbConfig;
+    try {
+      dbConfig =
+          DatabaseConfiguration.createConfigFromFile(dataProperties.getThriftserverHiveConfig());
+    } catch (IOException e) {
+      logger.error("Exception while reading thrift hive config.");
+      throw new RuntimeException(e);
+    }
+    HiveTableManager hiveTableManager =
+        new HiveTableManager(
+            dbConfig.makeJdbsUrlFromConfig(),
+            dbConfig.getDatabaseUser(),
+            dbConfig.getDatabasePassword());
+
+    String rootPrefix = dataProperties.getDwhRootPrefix();
+    Preconditions.checkState(rootPrefix != null && !rootPrefix.isEmpty());
+
+    String prefix = getPrefix(rootPrefix);
+    String baseDir = getBaseDir(rootPrefix);
+    List<ResourceId> paths;
+    try {
+      paths =
```

Comment:
I understand that this class has no unit-tests at the moment, but given our discussions and your roadmap, do you want to add some unit-tests for the new functionality in this PR? If you prefer to leave this for the near future, that is fine too, but I feel the logic you have added warrants some unit-tests.

Reply:
I will add unit test cases as part of increasing coverage as this is slightly different than usual.

Reply:
Sounds good.